server.json

```diff
@@ -0,0 +1,14 @@
+{
+  "extends": [
+    "./common.json"
+  ],
+  "env": {
+    "es6": true,
```

Comment:
I think we should document what minimum Node version the preset is aimed at. The ES6 grammar was first introduced in Node 4, with most of the actual classes/methods arriving in Node 6. I'd say this currently resembles a Node 6+ environment.

Maybe also document that if a project's minimum support version is lower, to extend `common` directly instead (treat as custom config, starting with the basic preset). If we encourage projects to extend the new preset instead, they'd have to disable numerous seemingly unrelated rules that don't work well in older environments. In the past this was sometimes a source of frustration for us with upstream presets, and it'd be nice to avoid that from the start. We could also revisit that later, but if it's not controversial, might be better to start with. Given that it is easier to add support later (drop the social restriction), than to remove support later (add a new social restriction).

Reply:
Yes, right now client is common+browser+es5 (current MediaWiki support) and server is common+node+es6 (current Wikimedia production node server version). When we break out the language support files that'll be more explicit, I think. Will document as such.